apps/aecore/src/aec_upnp.erl

```diff
@@ -66,18 +88,27 @@ handle_cast(Other, State) ->
     epoch_sync:warning("Received unknown cast: ~p", [Other]),
     {noreply, State}.
 
-handle_info(add_port_mapping, State) ->
-    proc_lib:spawn(fun() -> add_port_mapping() end),
+handle_info(add_port_mapping, State0) ->
+    ok = kill_mapping_worker(State0),
+    State = spawn_mapping_worker(),
+
     %% Give additional 10 secs for UPnP/NAT-PMP discovery and setup, to
     %% make sure there is continuity in port mapping.
     erlang:send_after(1000 * (?MAPPING_LIFETIME - 10), self(), add_port_mapping),
     {noreply, State};
+handle_info({'EXIT', _Pid, normal}, State) ->
+    %% Received when calling delete_port_mapping/0 during prep_stop
+    {noreply, State};
+handle_info({'DOWN', Ref, process, Pid, normal}, #state{pid = Pid, mon = Ref}) ->
+    {noreply, #state{}};
+handle_info({'DOWN', Ref, process, Pid, Why}, #state{pid = Pid, mon = Ref}) when Why =/= normal ->
+    epoch_sync:warning("UPnP/NAT-PMP worker finished: ~p", [Why]),
+    {noreply, #state{}};
 handle_info(Other, State) ->
     epoch_sync:warning("Received unknown info message: ~p", [Other]),
     {noreply, State}.
 
 terminate(_Reason, _State) ->
-    delete_port_mapping(),
```

Comment:
So why keeping the trapping of exits - with consequent `handle_info({'EXIT', _Pid, normal}, State)` clause?

Reply:
Good point, this is not needed anymore. I will update.